src/awx_plugins/credentials/github_app.py

```diff
@@ -0,0 +1,185 @@
+    'fields': [
+        {
+            'id': 'github_url',
+            'label': _('GitHub URL'),
+            'type': 'string',
+            'help_text': _(
+                'Specify the GitHub URL here in the case of an Enterprise Github. '
+                'Example: https://gh.your.org/api/v3',
+            ),
+            'default': 'https://api.github.com',
+        },
+        {
+            'id': 'app_id',
+            'label': _('GitHub App id created by the GitHub Admin'),
+            'type': 'string',
+            'help_text': _(
+                'The GitHub App ID created by the GitHub Admin. '
+                'Example App ID: 1121547 '
+                'from https://github.com/settings/apps/arestlel-github-app-ansible '
+                'required for creating a JWT token for authentication',
+            ),
+        },
+        {
+            'id': 'install_id',
+            'label': _('Installation ID'),
+            'type': 'string',
+            'help_text': _(
+                'The Installation ID from the GitHub App installation generated by the GitHub Admin. '
+                'Example: 59980338 '
+                'extracted from the installation link: https://github.com/settings/installations/59980338 '
+                'required for creating a limited GitHub app token',
+            ),
+        },
+        {
+            'id': 'private_rsa_key',
+            'label': _('RSA Private Key'),
+            'type': 'string',
+            'format': 'ssh_private_key',  # this validator checks for formats close to PKCS#1 PEMs
+            'secret': True,
+            'multiline': True,
+            'help_text': _(
+                'Paste the contents of the PEM file that the GitHub Admin '
+                'provided to you with the app and installation IDs.',
+            ),
+        },
+    ],
+    'metadata': [
+        {
+            'id': 'description',
+            'label': _('Description (Optional)'),
+            'type': 'string',
+            'help_text': _('To be removed after UI is updated'),
+        },
+    ],
+    'required': ['app_id', 'install_id', 'private_rsa_key'],
+}
+
+class GitHubAppBackendArgs(TypedDict, total=False):
+    github_url: str
+    app_id: str | int
+    install_id: str | int
+    private_rsa_key: str
+    jwt_expiry: str | int
+
+
+def github_app_backend(**kwargs: GitHubAppBackendArgs) -> str:
+    """Generate an authentication token for a GitHub App using provided
+    credentials.
+
+    Args:
+        github_url (str): The GitHub instance URL.
+        app_id (str): The GitHub App ID.
+        install_id (str): The installation ID for the GitHub App.
+        private_rsa_key (str): The private key associated with the GitHub App.
+        jwt_expiry (int, optional): JWT expiration time in seconds (default: 600).
+
+    Returns:
+        str: A GitHub App authentication token.
+
+    Raises:
+        ValueError: If any required parameters are missing or invalid.
```

Comment:
Semi-broken rendering: https://awx-plugins-core--84.org.readthedocs.build/en/84/pkg/awx_plugins.credentials.github_app/#awx_plugins.credentials.github_app.github_app_backend.
```suggestion
    :param github_url: The GitHub instance API URL.
    :type github_url: str

    :param app_id: The GitHub App ID.
    :type app_id: str

    :param private_rsa_key: The private key associated with the GitHub App.
    :type private_rsa_key: str

    :param jwt_expiry: JWT expiration time in seconds (default: 600).
    :type jwt_expiry: int | None

    :returns: A GitHub App authentication token.
    :rtype: str

    :raises ValueError: If any required parameters are missing or invalid.
```